pkg/querysharding/analyzer_test.go

```diff
@@ -200,3 +202,353 @@ http_requests_total`,
+	},
+	{
+		name:           "multiple binary expressions with grouping",
+		expression:     `(http_requests_total{code="400"} + on (pod) http_requests_total{code="500"}) / on (cluster, pod) http_requests_total`,
+		shardingLabels: []string{"cluster", "pod"},
+	},
+	{
+		name:           "histogram quantile",
+		expression:     "histogram_quantile(0.95, sum(rate(metric[1m])) by (le, cluster))",
+		shardingLabels: []string{"cluster"},
+	},
+	{
+		name: "multiple binary expressions with grouping",
+		expression: `
+sum by (container) (
+(http_requests_total{code="400"} + on (cluster, pod, container) http_requests_total{code="500"})
+/ on (pod, container)
+http_requests_total
+)`,
+		shardingLabels: []string{"container"},
+	},
+	{
+		name:           "multiple binary expressions with grouping",
+		expression:     `(http_requests_total{code="400"} + on (pod) http_requests_total{code="500"}) / on (cluster, pod) http_requests_total`,
+		shardingLabels: []string{"cluster", "pod"},
+	},
+	{
+		name:           "histogram quantile",
+		expression:     "histogram_quantile(0.95, sum(rate(metric[1m])) by (le, cluster))",
+		shardingLabels: []string{"cluster"},
+	},
+	{
+		name: "multiple binary expressions with grouping",
+		expression: `
+sum by (container) (
+(http_requests_total{code="400"} + on (cluster, pod, container) http_requests_total{code="500"})
+/ on (pod, container)
+http_requests_total
+)`,
+		shardingLabels: []string{"container"},
+	},
+	{
+		name:           "multiple binary expressions with grouping",
+		expression:     `(http_requests_total{code="400"} + on (pod) http_requests_total{code="500"}) / on (cluster, pod) http_requests_total`,
+		shardingLabels: []string{"cluster", "pod"},
+	},
+	{
+		name:           "histogram quantile",
+		expression:     "histogram_quantile(0.95, sum(rate(metric[1m])) by (le, cluster))",
+		shardingLabels: []string{"cluster"},
+	},
+	{
+		name: "multiple binary expressions with grouping",
+		expression: `
+sum by (container) (
+(http_requests_total{code="400"} + on (cluster, pod, container) http_requests_total{code="500"})
+/ on (pod, container)
+http_requests_total
+)`,
+		shardingLabels: []string{"container"},
+	},
+	{
+		name:           "multiple binary expressions with grouping",
+		expression:     `(http_requests_total{code="400"} + on (pod) http_requests_total{code="500"}) / on (cluster, pod) http_requests_total`,
+		shardingLabels: []string{"cluster", "pod"},
+	},
+	{
+		name:           "histogram quantile",
+		expression:     "histogram_quantile(0.95, sum(rate(metric[1m])) by (le, cluster))",
+		shardingLabels: []string{"cluster"},
+	},
+}
+
+func BenchmarkTestAnalyzeQuery(b *testing.B) {
+	analyze := NewQueryAnalyzer()
+	for _, v := range value {
+		b.Run(fmt.Sprintf("query: %s", v.name), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				_, err := analyze.Analyze(v.expression)
+
+				if err != nil {
+					fmt.Printf("Error while benchmarking =%s ", err)
+				}
+			}
+		})
+	}
+}
```

Comment:
Let's remove the benchmark and the `value` global vars. Currently no need to have this.